transport/httptransport/libp2p_server.go

```diff
@@ -223,6 +233,10 @@ func (s *Libp2pCarServer) sendCar(r *http.Request, w http.ResponseWriter, val *A
 		err = e
 	}}
 
+	// Get a channel that will be closed when the client closes the connection
+	stream := getConn(r).(gostream.Stream)
+	closeCh := s.streamMonitor.getCloseChan(stream.ID())
```

Comment:
This is racy .

If there's a disconnection while serving the file below, you will deem success on a stream that the client isn't gonna close anymore or has already closed because that is from the previous connection.

Reply:
I'm really leaning towards simply enquiring the deal status to know that it has finished transferring rather than doing the state management gymnastics here.

Reply:
If there's a disconnect during the transfer, the err will be non-nil and we will return the err on line 249